source/extensions/filters/listener/tls_inspector/tls_inspector.cc

```diff
@@ -235,6 +244,149 @@ ParseState Filter::parseClientHello(const void* data, size_t len) {
   }
 }
 
+static const uint16_t GREASE[] = {
```

Comment:
I think this PR would benefit from an explanation of what TLS client fingerprinting is. Presumably it's just a hash of some subset of fields in the TLS client hello. Which makes sense, I'm aware of some applications of this. I'd like to understand here the rationale behind choice of fields and what purpose this is intended to serve. This should be done in both the API and also more comments added across the code here.
/wait

Reply:
e.g. what is grease?

Reply:
Thanks, I wasn't sure how much context this needed -- your description is accurate, as for the the rationale behind choice of fields and the choice of md5 as a hashing algorithm, this is specified by the JA3 standard, which allows the fingerprints to be shared/recorded in threat databases, etc.

Reply:
[GREASE (Generate Random Extensions And Sustain Extensibility)](https://datatracker.ietf.org/doc/html/draft-davidben-tls-grease-01) is a mechanism Google uses to prevent extensibility failures in the TLS ecosystem. JA3 ignores these values completely to ensure that programs utilizing GREASE can still be identified with a single JA3 hash.